Java/core/src/main/java/com/amazon/randomcutforest/tree/CompactNodeManager.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.tree;
+
+import static com.amazon.randomcutforest.CommonUtils.checkState;
+
+import com.amazon.randomcutforest.store.ILeafStore;
+import com.amazon.randomcutforest.store.INodeStore;
+import com.amazon.randomcutforest.store.LeafStore;
+import com.amazon.randomcutforest.store.NodeStore;
+
+/**
+ * An interface for describing access to the node store for different types of
+ * trees.
+ */
+public class CompactNodeManager {
```

Comment:
The mixing of `null` and `NULL` makes this code hard to understand. We should try to reduce or eliminate the use of `null`. I'll create an issue to follow up after this PR.

#104

Reply:
makes sense. We need the pointer type and null to support PointerTrees (or equivalent name)